src/compiler/checker.ts

```diff
@@ -2756,6 +2764,10 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
                     return symbol;
                 }
                 if (symbol.flags & SymbolFlags.Alias) {
+                    // Do not take target symbol meaning into account in isolated declaration mode since we don't have access to info from other files.
+                    if (compilerOptions.isolatedDeclarations) {
+                        return symbol;
+                    }

```

Comment:
I reviewed the tests that would fail without this. It mostly comes down to TSC removing some imports that are not used based on semantic info. 


Ex: An import is a type, but the same name also refers to a global, and the global is what si actually used in the declaration.

```ts
// TSC will remove this import, DTE can't know to remove it
import { $ } from "./ref"; // $ is a type actually, there is also a global $
export interface A {
    x: () => typeof $; // the global $ is used. not the imported $
}
```

This seems fine as DTE/TSC difference. See https://github.com/bloomberg/TypeScript/pull/121 for full list of impacted tests.